aws/resource_aws_batch_job_queue.go

```diff
@@ -0,0 +1,205 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/batch"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsBatchJobQueue() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsBatchJobQueueCreate,
+		Read:   resourceAwsBatchJobQueueRead,
+		Update: resourceAwsBatchJobQueueUpdate,
+		Delete: resourceAwsBatchJobQueueDelete,
+
+		Schema: map[string]*schema.Schema{
+			"compute_environments": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"priority": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"state": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{batch.JQStateDisabled, batch.JQStateEnabled}, true),
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsBatchJobQueueCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).batchconn
+	input := batch.CreateJobQueueInput{
+		ComputeEnvironmentOrder: createComputeEnvironmentOrder(d),
+		JobQueueName:            aws.String(d.Get("name").(string)),
+		Priority:                aws.Int64(int64(d.Get("priority").(int))),
+		State:                   aws.String(d.Get("state").(string)),
+	}
+	out, err := conn.CreateJobQueue(&input)
+	name := d.Get("name").(string)
```

Comment:
NIT: Consider getting the name first, so the error check follows immediately the call to `conn.CreateJobQueue`